packages/data/src/components/use-dispatch/test/use-dispatch.js

```diff
@@ -11,165 +12,123 @@ import createReduxStore from '../../../redux-store';
-			return <button onClick={ foo } />;
+			const { inc } = useDispatch( 'demoStore' );
+			return <button onClick={ inc } />;
 		};
 
-		let testRenderer;
-
-		act( () => {
-			testRenderer = TestRenderer.create(
-				<RegistryProvider value={ registry }>
-					<TestComponent />
-				</RegistryProvider>
-			);
-		} );
-
-		const testInstance = testRenderer.root;
+		render(
+			<RegistryProvider value={ registry }>
+				<TestComponent />
+			</RegistryProvider>
+		);
 
-		act( () => {
-			testInstance.findByType( 'button' ).props.onClick();
-		} );
+		await user.click( screen.getByRole( 'button' ) );
 
-		expect( testAction ).toHaveBeenCalledTimes( 1 );
+		expect( registry.select( 'demoStore' ).get() ).toBe( 1 );
 	} );
 
-	it( 'returns dispatch from correct registry if registries change', () => {
-		const reducer = ( state ) => state;
-		const noop = () => ( { type: '__INERT__' } );
-		const firstRegistryAction = jest.fn().mockImplementation( noop );
-		const secondRegistryAction = jest.fn().mockImplementation( noop );
+	it( 'returns dispatch from correct registry if registries change', async () => {
+		const user = userEvent.setup();
 
-		const firstRegistry = registry;
-		firstRegistry.registerStore( 'demo', {
-			reducer,
-			actions: {
-				noop: firstRegistryAction,
-			},
-		} );
+		const firstRegistry = createRegistry();
+		firstRegistry.registerStore( 'demo', counterStore );
+
+		const secondRegistry = createRegistry();
+		secondRegistry.registerStore( 'demo', counterStore );
 
 		const TestComponent = () => {
 			const dispatch = useDispatch();
-			return <button onClick={ () => dispatch( 'demo' ).noop() } />;
+			return <button onClick={ () => dispatch( 'demo' ).inc() } />;
 		};
 
-		let testRenderer;
-		act( () => {
-			testRenderer = TestRenderer.create(
-				<RegistryProvider value={ firstRegistry }>
-					<TestComponent />
-				</RegistryProvider>
-			);
-		} );
-		const testInstance = testRenderer.root;
+		const { rerender } = render(
+			<RegistryProvider value={ firstRegistry }>
+				<TestComponent />
+			</RegistryProvider>
+		);
 
-		act( () => {
-			testInstance.findByType( 'button' ).props.onClick();
-		} );
+		await user.click( screen.getByRole( 'button' ) );
 
-		expect( firstRegistryAction ).toHaveBeenCalledTimes( 1 );
-		expect( secondRegistryAction ).toHaveBeenCalledTimes( 0 );
+		expect( firstRegistry.select( 'demo' ).get() ).toBe( 1 );
+		expect( secondRegistry.select( 'demo' ).get() ).toBe( 0 );
 
-		const secondRegistry = createRegistry();
-		secondRegistry.registerStore( 'demo', {
-			reducer,
-			actions: {
-				noop: secondRegistryAction,
-			},
-		} );
-
-		act( () => {
-			testRenderer.update(
-				<RegistryProvider value={ secondRegistry }>
-					<TestComponent />
-				</RegistryProvider>
-			);
-		} );
-		act( () => {
-			testInstance.findByType( 'button' ).props.onClick();
-		} );
-		expect( firstRegistryAction ).toHaveBeenCalledTimes( 1 );
-		expect( secondRegistryAction ).toHaveBeenCalledTimes( 1 );
```

Comment:
Yup 👍 I agree with that. The only reason why I'd want to see us specifically dispatching the right action is because it's actually `useDispatch` we're testing, so we want to ensure that it's dispatching correctly. Also it would make things more explicit. But I agree that from the user perspective it's enough to test the result store values, so feel free to keep it as-is, I have no strong feelings either way.

Reply:
Added new test for this case - [f6c1a3c](https://github.com/WordPress/gutenberg/pull/44802/commits/f6c1a3c672e40cfdf375bf981852d6ae45ab0d7b).